### PR TITLE
Transitions/modals

### PR DESCRIPTION
**Pull request by EstebanIndiveri**
## Summary
Added new StackScreen Gallery and modal
Transition by navigation and Animated API,
tested component detailPost

## Screenshots

[Change!] Show the screenshots of the views you modified.

### Android
<a href="https://ibb.co/KDwmyj3">Screenshot-1621350142</a>

#### Galaxy S (or similar)

<a href="https://ibb.co/18ZX7H0">Screenshot-1621350092</a>
<a href="https://ibb.co/5Bqzdxv">Screenshot-1621350112</a>
<a href="https://ibb.co/pPg57DP">Screenshot-1621350125</a><br /><a target='_blank' href='https://es.imgbb.com/'>web png</a><br />
### iOS
<a href="https://ibb.co/WkyMvTK">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-18-at-12-05-00</a>

<a href="https://ibb.co/WpDCd0X">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-18-at-12-05-10</a>

<a href="https://ibb.co/swZ1xT8">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-18-at-12-05-21</a>

#### iPhone SE (or similar)
<a href="https://ibb.co/7C5DXFr">Simulator-Screen-Shot-i-Phone-8-2021-05-18-at-12-03-45</a>

<a href="https://ibb.co/nQZknRF">Simulator-Screen-Shot-i-Phone-8-2021-05-18-at-12-03-57</a>

<a href="https://ibb.co/xFM5M28">Simulator-Screen-Shot-i-Phone-8-2021-05-18-at-12-04-07</a>

#### iPhone 6/7/8 (or similar)
<a href="https://ibb.co/6DTfRtb">Simulator-Screen-Shot-i-Phone-12-2021-05-18-at-12-05-50</a>

<a href='https://postimg.cc/XZs9n0KM' target='_blank'>Simulator-Screen-Recording-i-Phone-12-2021-05-18-at-12-17-13</a>

## Trello Card
[Card Trello Link](https://trello.com/c/fjPysycl/35-modales-dialogs)

### CODE REVIEW

File: wbooks/src/app/navigator/index.tsx [hunk 1]
```diff
@@ -12,12 +12,17 @@ import {
...
   DETAIL_BOOK_OPTIONS,
   HOME_OPTIONS,
-  STACK_NAVIGATOR_OPTIONS
+  STACK_NAVIGATOR_OPTIONS,
+  SHARED_NAVIGATOR_OPTIONS
 } from '@constants/screenOptions';
 import { NavigationContainer } from '@react-navigation/native';
+import Gallery from '@screens/Gallery';
+import DetailPost from '@screens/DetailPost';
+import { createSharedElementStackNavigator } from 'react-navigation-shared-element';
```
Comment (mattgle) [inline wbooks/src/app/navigator/index.tsx]: this should be on line 18

File: wbooks/src/app/screens/DetailPost/index.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useEffect } from 'react';
+import { View, Image, Text, Animated, TouchableWithoutFeedback, ScrollView } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import IC_BACK from '@assets/General/ic_back.png';
+import { PostsRoutes } from '@interfaces/posts';
+import { NavigationProp, RouteProp } from '@react-navigation/native';
+import { SharedElement } from 'react-navigation-shared-element';
```
Comment (mattgle) [inline wbooks/src/app/screens/DetailPost/index.tsx]: this 2 should be at line 4

File: wbooks/src/app/screens/DetailPost/index.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import styles from './styles';
+
+interface Props {
+  navigation: NavigationProp<PostsRoutes, 'post'>;
```
Comment (mattgle) [inline wbooks/src/app/screens/DetailPost/index.tsx]: extract 'post'

File: wbooks/src/app/screens/DetailPost/index.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  };
+
+  return (
+    <ScrollView showsVerticalScrollIndicator={false} style={styles.wrapper}>
+      <Animated.View style={[styles.container, customStyles]}>
+        <TouchableWithoutFeedback onPress={handlePress}>
+          <Image style={styles.icon} resizeMethod="resize" resizeMode="contain" source={IC_BACK} />
+        </TouchableWithoutFeedback>
+      </Animated.View>
+      <SharedElement id={post.id}>
```
Comment (mattgle) [inline wbooks/src/app/screens/DetailPost/index.tsx]: destructure post

File: wbooks/src/app/screens/DetailPost/styles.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import { cyan, white } from '@constants/colors';
+
+const { StyleSheet } = require('react-native');
```
Comment (mattgle) [inline wbooks/src/app/screens/DetailPost/styles.js]: why require? use import

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { useNavigation } from '@react-navigation/native';
+import React from 'react';
+import { SafeAreaView, ScrollView, Pressable, Image, View, Text } from 'react-native';
+import { Mockposts } from '@constants/mockposts';
+import Routes from '@constants/routes';
+import { SharedElement } from 'react-navigation-shared-element';
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx]: native import with native imports pls

**Review by mattgle — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/Gallery/styles.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    margin: 15
+  },
+  posts: {
+    flex: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap'
+  },
+  postTexts: {
+    margin: 10,
+    marginBottom: 15
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/screens/Gallery/styles.js:22]: use const to same values

File: wbooks/src/app/navigator/index.tsx [hunk 8]
```diff
@@ -12,12 +12,17 @@ import {
   TAB_BAR_OPTIONS,
   DETAIL_BOOK_OPTIONS,
   HOME_OPTIONS,
-  STACK_NAVIGATOR_OPTIONS
+  STACK_NAVIGATOR_OPTIONS,
+  SHARED_NAVIGATOR_OPTIONS
 } from '@constants/screenOptions';
+import Gallery from '@screens/Gallery';
+import DetailPost from '@screens/DetailPost';
+import { createSharedElementStackNavigator } from 'react-navigation-shared-element';
```
Comment (mattgle) [inline wbooks/src/app/navigator/index.tsx]: This import is an import from node_modules, so it should be next to those. In this case you should move this lineto line 4.

File: wbooks/src/app/navigator/index.tsx [hunk 9]
```diff
@@ -12,12 +12,17 @@ import {
...
 import { NavigationContainer } from '@react-navigation/native';
```
Comment (mattgle) [inline wbooks/src/app/navigator/index.tsx]: same comment

File: wbooks/src/app/screens/DetailPost/index.tsx (hunk 3 above)
Comment (mattgle) [reply wbooks/src/app/screens/DetailPost/index.tsx]: ...

File: wbooks/src/app/screens/DetailPost/styles.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+const BOLD = 'bold';
+const ONEHUNDRERPERCENT = '100%';
+const CENTER = 'center';
+const ROW = 'row';
+
+const ONE = 1;
+const TEN = 10;
+const SIXTY = 60;
+const TWENTYFOUR = 24;
+const TWENTYFIVE = 25;
```
Comment (mattgle) [inline wbooks/src/app/screens/DetailPost/styles.js]: ? What happend to the comment i made about this?

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        <SharedElement id={post.id}>
+          <Image source={{ uri: post.source }} style={styles.image} resizeMethod="scale" resizeMode="cover" />
+        </SharedElement>
+        <View style={styles.postTexts}>
+          <Text numberOfLines={1} style={styles.postHeader}>
+            {post.user}
+          </Text>
+        </View>
+      </Pressable>
+    ));
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx]: destructure post

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  return (
+    <SafeAreaView style={styles.wrapper}>
+      <ScrollView style={styles.wrapper} showsHorizontalScrollIndicator={false}>
+        <Text style={styles.listHeader}>{GREAT_PHOTOS}</Text>
+        <View style={styles.posts}>{renderMockPosts()}</View>
+      </ScrollView>
+    </SafeAreaView>
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx]: is thi SafeAreaView needed?

File: wbooks/src/app/screens/Gallery/styles.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
+const { StyleSheet } = require('react-native');
+
+const FONT_BOLD = 'bold';
+const FIFTEEN = 15;
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/styles.js]: same comment.

File: wbooks/src/app/screens/Gallery/styles.js [hunk 14]
```diff
@@ -0,0 +1,36 @@
+const { StyleSheet } = require('react-native');
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/styles.js]: why require? use imports.

File: wbooks/src/interfaces/posts.ts [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+  id: string;
+  source: string;
+  user: string;
+  avatar: string;
+  video?: string;
+}
+
+export type PostsRoutes = {
+  routes: undefined;
+  post: { post: Post };
```
Comment (mattgle) [inline wbooks/src/interfaces/posts.ts]: this is weird. Why post: {post: } Cant you change the name? and why routes is undefined?

**Review by mattgle — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+  function renderMockPosts() {
+    const handlePress = (id: string, source: string, avatar: string, user: string) => {
+      navigation.navigate(Routes.DetailPost, {
+        post: { id, source, avatar, user }
+      });
+    };
+    return Mockposts.map(({ id, source, avatar, user }: Post) => (
+      <Pressable
+        key={id}
+        onPress={() => handlePress(id, source, avatar, user)}
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx]:
```suggestion
        onPress={handlePress(id, source, avatar, user)}
```

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+export default function Gallery() {
+  const navigation = useNavigation();
+  const GREAT_PHOTOS = 'Great Photos';
+
+  function renderMockPosts() {
+    const handlePress = (id: string, source: string, avatar: string, user: string) => {
+      navigation.navigate(Routes.DetailPost, {
+        post: { id, source, avatar, user }
+      });
+    };
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx]:
```suggestion
    const handlePress = (id: string, source: string, avatar: string, user: string)  = () => {
      navigation.navigate(Routes.DetailPost, {
        post: { id, source, avatar, user }
      });
    };
```

File: wbooks/src/interfaces/posts.ts [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+  avatar: string;
+  video?: string;
+}
+type RootStackParamList = {
+  Gallery: Routes.Gallery;
+  DetailPost: Routes.DetailPost;
+};
+
+export type PostsRoutes = {
+  routes: RootStackParamList;
```
Comment (mattgle) [inline wbooks/src/interfaces/posts.ts]: Cant you just use Routes?, i Mean its an enum

**Review by mattgle — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/Gallery/index.tsx [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+import { Post } from '@interfaces/posts';
+
+import styles from './styles';
+
+export default function Gallery() {
+  const navigation = useNavigation();
+  const GREAT_PHOTOS = 'Great Photos';
+
+  function renderMockPosts() {
+    const handlePress = (id: string, source: string, avatar: string, user: string) => () => {
```
Comment (mattgle) [inline wbooks/src/app/screens/Gallery/index.tsx:16]: why dont you use an object, and use the Post interface for props object?

**Review by mattgle — APPROVED** [review]

**Pull request merged**